src/core_atmosphere/dynamics/mpas_atm_time_integration.F

```diff
@@ -927,7 +932,17 @@ subroutine atm_srk3(domain, dt, itimestep, exchange_halo_group)
                   if (index_ni > 0) then
                      scalars_driving(index_ni,1:nVertLevels,1:nCells+1) = mpas_atm_get_bdy_state( clock, block, nVertLevels, nCells, 'ni', rk_timestep(rk_step) )
                   end if
-                  
+                  if(config_microp_scheme == 'mp_thompson_aerosols') then
+                     if (index_nc > 0) then
+                        scalars_driving(index_nc,1:nVertLevels,1:nCells+1) = mpas_atm_get_bdy_state( clock, block, nVertLevels, nCells, 'ni', rk_timestep(rk_step) )
+                     end if
+                     if (index_nifa > 0) then
+                        scalars_driving(index_nifa,1:nVertLevels,1:nCells+1) = mpas_atm_get_bdy_state( clock, block, nVertLevels, nCells, 'ni', rk_timestep(rk_step) )
+                     end if
+                     if (index_nwfa > 0) then
+                        scalars_driving(index_nwfa,1:nVertLevels,1:nCells+1) = mpas_atm_get_bdy_state( clock, block, nVertLevels, nCells, 'ni', rk_timestep(rk_step) )
```

Comment:
I think the second-to-last arguments aren't correct, and they should be `nc`, `nifa`, and `nwfa`.

Reply:
you are right. corrected.